### PR TITLE
CONTRIBUTING.md (Open for Discussion!)

### PR DESCRIPTION
**Pull request by codydaig**
I went ahead and created a CONTRIBUTING.md based off of puppet's. These were just my initial thoughts, and completely open for discussion! 

(*NOTE: I think has to be merged in with master to show up on the pull request's page. *)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	* If possible, please "squash" your commits to as few commits as possible/reasonable.
+* Make Sure you have added any tests necessary to test your code.
+	* Run __all__ the tests to ensure nothing else was acciddently broken.
+	* Don't rely on the existing tests to see if you've broken code elsewhere, test the changes you made in a browser too!
+* Update the Documentation too go along with any changes in functionality / improvements in a seperate pull request.
+
+## Submitting the Pull Request
+* Push your changes to your topic branch on your fork of the repo.
+* Submit a pull request from your topic branch to the necessary branch on the MEANJS repository.
+* Be sure to tag any issues your pull request is taking care of / contributing too.
```
Comment (cdriscol) [inline CONTRIBUTING.md]: Grammer-:cop: : 'too' should be 'to'

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+* Create a topic branch from where you want to base your work. 
+	* Please note that branches of older versions are for critical bugs only.
+* Check for unnecessary whitespace / changes with `git diff --check` before committing.
+	* Also check that your code is formatted properly with tabs (hint: Use .editorconfig)
+* Keep git commit messages clear and appropriate
+	* If possible, please "squash" your commits to as few commits as possible/reasonable.
+* Make Sure you have added any tests necessary to test your code.
+	* Run __all__ the tests to ensure nothing else was acciddently broken.
+	* Don't rely on the existing tests to see if you've broken code elsewhere, test the changes you made in a browser too!
```
Comment (cdriscol) [inline CONTRIBUTING.md]: Grammer-:cop: : 'too' should be 'to'

**Comment by cdriscol** [comment]
:+1:

**Comment by codydaig** [comment]
@cdriscol Fixed! :-D Thanks!

**Comment by codydaig** [comment]
@lirantal I made the changes to reflect the new branch strategy.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+* Clearly describe the issue, including the steps to reproduce the issue.
+	* If it's a new feature, enhancement, or restructure, Explain your reasonings on why you think it should be added, as well as a particular use case.
+
+## Making Changes
+
+* Create a topic branch from the master branch.
+* Check for unnecessary whitespace / changes with `git diff --check` before committing.
+	* Also check that your code is formatted properly with spaces (hint: Use .editorconfig)
+* Keep git commit messages clear and appropriate
+	* If possible, please "squash" your commits to as few commits as possible/reasonable.
```
Comment (ilanbiala) [inline CONTRIBUTING.md]: I would make it more specific go squash into reasonable separations such as one commit for implementation, one for tests, one for documentation before finally squashing to one commit when getting the LGTM from a collaborator.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+* Make Sure you have added any tests necessary to test your code.
+	* Run __all__ the tests to ensure nothing else was acciddently broken.
+	* Don't rely on the existing tests to see if you've broken code elsewhere, test the changes you made in a browser to!
```
Comment (ilanbiala) [inline CONTRIBUTING.md]: Too

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+* Update the Documentation too go along with any changes in functionality / improvements in a seperate pull request against the gh-pages branch.
```
Comment (ilanbiala) [inline CONTRIBUTING.md]: To

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+## Submitting the Pull Request
+* Push your changes to your topic branch on your fork of the repo.
+* Submit a pull request from your topic branch to the master branch on the MEANJS repository.
+* Be sure to tag any issues your pull request is taking care of / contributing to.
```
Comment (ilanbiala) [inline CONTRIBUTING.md]: "Fixes #xyz" is helpful because it will close that issue for us.

**Comment by codydaig** [comment]
@ilanbiala Made the requested changes!

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+* Keep git commit messages clear and appropriate
+	* If possible, please "squash" your commits to as few commits as possible/reasonable such as one commit for implementation, one for tests, and one for documentation before finally squashing to one commit when getting the LGTM from a collaborator.
+* Make Sure you have added any tests necessary to test your code.
+	* Run __all__ the tests to ensure nothing else was acciddently broken.
+	* Don't rely on the existing tests to see if you've broken code elsewhere, test the changes you made in a browser too!
+* Update the Documentation to go along with any changes in functionality / improvements in a seperate pull request against the gh-pages branch.
+
+## Submitting the Pull Request
+* Push your changes to your topic branch on your fork of the repo.
+* Submit a pull request from your topic branch to the master branch on the MEANJS repository.
```
Comment (ilanbiala) [inline CONTRIBUTING.md]: MEAN.js

**Comment by codydaig** [comment]
@ilanbiala Made the latest requested change.

**Comment by codydaig** [comment]
Closing in favor of #815

**Pull request closed without merging**